src/System.Linq.Expressions/src/System/Linq/Expressions/CatchBlock.cs

```diff
@@ -158,7 +157,20 @@ public static CatchBlock MakeCatchBlock(Type type, ParameterExpression variable,
         {
             ContractUtils.RequiresNotNull(type, nameof(type));
             ContractUtils.Requires(variable == null || TypeUtils.AreEquivalent(variable.Type, type), nameof(variable));
-            if (variable != null && variable.IsByRef)
+            if (variable == null)
+            {
+                TypeUtils.ValidateType(type, nameof(type));
```

Comment:
At the very least they could go through a single method that does all three calls. Other PRs for the same issue have the same pattern with the same opportunity to reduce redundancy. I'm planning to tie them all together when I've all the places such types can sneak in plugged.